snapcraft/internal/project_loader.py

```diff
@@ -128,188 +122,33 @@ def __init__(self, project_options=None):
 
         self.build_tools = self.data.get('build-packages', [])
         self.build_tools.extend(project_options.additional_build_packages)
-        self._process_parts()
 
-    @property
-    def part_names(self):
-        return self._part_names
-
-    @property
-    def _remote_parts(self):
-        if getattr(self, '_remote_parts_attr', None) is None:
-            self._remote_parts_attr = parts.get_remote_parts()
-        return self._remote_parts_attr
-
-    def _process_parts(self):
-        for part_name in self.data.get('parts', []):
-            self._part_names.append(part_name)
-            properties = self.data['parts'][part_name] or {}
-
-            plugin_name = properties.pop('plugin', None)
-            if not plugin_name:
-                logger.info(
-                    'Searching in the remote parts cache for part '
-                    '{!r}'.format(part_name))
-                try:
-                    properties = self._remote_parts.compose(
-                        part_name, properties)
-                    plugin_name = properties.pop('plugin', None)
-                except KeyError as e:
-                    raise SnapcraftLogicError(
-                        '{!r} is missing the `plugin` entry and is not '
-                        'defined in the current remote parts cache, try to '
-                        'run `snapcraft update` to '
-                        'refresh'.format(part_name)) from e
-
-            if 'after' in properties:
-                self.after_requests[part_name] = properties.pop('after')
-
-            properties['stage'] = _expand_filesets_for('stage', properties)
-            properties['snap'] = _expand_filesets_for('snap', properties)
-
-            if 'filesets' in properties:
-                del properties['filesets']
-
-            self.load_plugin(part_name, plugin_name, properties)
-
-        self._compute_part_dependencies()
-        self.all_parts = self._sort_parts()
+        self.parts_config = parts.PartsConfig(self.data.get('parts', {}),
```

Comment:
would it be weird to call this one only parts? So you call it like config.parts instead of config.parts_config ?